apps/api-extractor/src/generators/DtsRollupGenerator.ts

```diff
@@ -0,0 +1,311 @@
+
+    const previousSpan: Span | undefined = span.previousSibling;
+
+    let recurseChildren: boolean = true;
+    switch (span.kind) {
+      case ts.SyntaxKind.JSDocComment:
+        // If the @packagedocumentation comment seems to be attached to one of the regular API items,
+        // omit it.  It gets explictly emitted at the top of the file.
+        if (span.node.getText().match(/(?:\s|\*)@packagedocumentation(?:\s|\*)/g)) {
+          span.modification.skipAll();
+        }
+
+        // For now, we don't transform JSDoc comment nodes at all
+        recurseChildren = false;
+        break;
+
+      case ts.SyntaxKind.ExportKeyword:
+      case ts.SyntaxKind.DefaultKeyword:
+      case ts.SyntaxKind.DeclareKeyword:
+        // Delete any explicit "export" or "declare" keywords -- we will re-add them below
+        span.modification.skipAll();
+        break;
+
+      case ts.SyntaxKind.InterfaceKeyword:
+      case ts.SyntaxKind.ClassKeyword:
+      case ts.SyntaxKind.EnumKeyword:
+      case ts.SyntaxKind.NamespaceKeyword:
+      case ts.SyntaxKind.ModuleKeyword:
+      case ts.SyntaxKind.TypeKeyword:
+      case ts.SyntaxKind.FunctionKeyword:
+        // Replace the stuff we possibly deleted above
+        let replacedModifiers: string = '';
+
+        // Add a declare statement for root declarations (but not for nested declarations)
+        if (!astDeclaration.parent) {
+          replacedModifiers += 'declare ';
+        }
+
+        if (entity.exported) {
+          replacedModifiers = 'export ' + replacedModifiers;
+        }
+
+        if (previousSpan && previousSpan.kind === ts.SyntaxKind.SyntaxList) {
+          // If there is a previous span of type SyntaxList, then apply it before any other modifiers
+          // (e.g. "abstract") that appear there.
+          previousSpan.modification.prefix = replacedModifiers + previousSpan.modification.prefix;
+        } else {
+          // Otherwise just stick it in front of this span
+          span.modification.prefix = replacedModifiers + span.modification.prefix;
+        }
+        break;
+
+      case ts.SyntaxKind.VariableDeclaration:
+        if (!span.parent) {
+          // The VariableDeclaration node is part of a VariableDeclarationList, however
+          // the Entry.followedSymbol points to the VariableDeclaration part because
+          // multiple definitions might share the same VariableDeclarationList.
+          //
+          // Since we are emitting a separate declaration for each one, we need to look upwards
+          // in the ts.Node tree and write a copy of the enclosing VariableDeclarationList
+          // content (e.g. "var" from "var x=1, y=2").
+          const list: ts.VariableDeclarationList | undefined = TypeScriptHelpers.matchAncestor(span.node,
+            [ts.SyntaxKind.VariableDeclarationList, ts.SyntaxKind.VariableDeclaration]);
+          if (!list) {
+            throw new Error('Unsupported variable declaration');
+          }
```

Comment:
I assume this is unexpected.

Reply:
This would happen only if the compiler API changes.

@iclanton I've been wanting to introduce a **node-core-library** class called something like "InternalError" whose handler will explain that an assumption was violated, and instruct the end user to open a GitHub issue to report the bug.

---
In reply to: [236473371](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236473371) [](ancestors = 236473371)

Reply:
Fixed

---
In reply to: [236583034](https://github.com/Microsoft/web-build-tools/pull/952#discussion_r236583034) [](ancestors = 236583034,236473371)